encoding/yaml/State.ts

```diff
@@ -0,0 +1,6 @@
+import { SchemaDefinition } from "./Schema.ts";
+import { DEFAULT_FULL_SCHEMA } from "./schema/mod.ts";
+
+export abstract class State {
+  constructor(public schema: SchemaDefinition = DEFAULT_FULL_SCHEMA) {}
+}
```

Comment:
I like that these objects are so well organized but... splitting them into 4 files means there are 4 more deps to download. I'd consider putting them into single file.

Reply:
Do you mean all 4 kinds of schema to be grouped as one file ?